### ISSUE TITLE
Calabash fails to fail when running a pipeline with an unbound primary input port (or so it seems)

### ISSUE THREAD

**Issue by vojtechtoman** [ndw/xmlcalabash1#55]
(Calabash 1.0.3-93)

Given the following pipeline (pipeline.xpl):

``` xml
<p:pipeline xmlns:p="http://www.w3.org/ns/xproc" version="1.0">
  <p:identity>
    <p:input port="source">
      <p:inline><doc/></p:inline>
    </p:input>
  </p:identity>
</p:pipeline>
```

If I do:

``` bash
java -jar calabash.jar pipeline.xpl
```

then the pipeline executes with no problems and produces the result:

``` xml
<doc/>
```

Unless there is some magic going on in Calabash, I think this is wrong because the primary input port or the pipeline is inbound. Calabash should either read the primary input from stdin, or it should fail with a dynamic error.

**Comment by ndw** [comment]
Looks like a bug to me. I've been doing some work on XML Calabash recently. I'll plan to get this fixed before the next release.

**Comment by ndw** [comment]
That's tricky.

It wouldn't be an error to run this pipeline with no source binding:

``` xml
<p:declare-step xmlns:p="http://www.w3.org/ns/xproc" version="1.0">
  <p:input port="source" primary="true">
    <p:inline>
      <defdoc/>
    </p:inline>
  </p:input>
  <p:output port="result"/>

  <p:identity>
    <p:input port="source">
      <p:inline><doc/></p:inline>
    </p:input>
  </p:identity>

</p:declare-step>
```

Would it be wrong to run this one:

``` xml
<p:declare-step xmlns:p="http://www.w3.org/ns/xproc" version="1.0">
  <p:input port="source" primary="true" sequence="true"/>
  <p:output port="result"/>

  <p:identity>
    <p:input port="source">
      <p:inline><doc/></p:inline>
    </p:input>
  </p:identity>

</p:declare-step>
```

It's a little hard to see where/how to address this bug.

**Comment by vojtechtoman** [comment]
The way I dealt with an unbound input port in the command-line interface of Calumet:
- Default binding exists?
  - Yes: use the default binding
  - No: is the port the primary input port?
    - Yes: read from stdin
  - No: fail

In other words, the first pipeline will execute with no problems (there is a default binding on the "source" input port), the latter will block and read input from stdin.

(: ...which makes me realize, somewhat embarrassingly, that I haven't implemented an explicit option to bind an input port to an empty sequence (needed also for overriding the default binding with an empty sequence) in Calumet. :)

So the difference in thinking is that you would like to provide more convenience and bind input ports declared as `sequence="true"` to an empty sequence by default, whereas in Calumet, users have to be explicit (they must provide bindings for unbound ports to avoid errors or the processor waiting on stdin).

**Comment by ndw** [comment]
Well. It turns out that the fix I just made for issue #39 makes this moot. If there's no binding for the primary input port, it will automatically be bound to STDIN.

That's not a fix, exactly, but...it does hide the bug :-)
